crates/aptos-crypto-derive/src/lib.rs

```diff
@@ -151,8 +151,12 @@ pub fn deserialize_key(source: TokenStream) -> TokenStream {
     let ast: DeriveInput = syn::parse(source).expect("Incorrect macro input");
     let name = &ast.ident;
     let name_string = name.to_string();
+    let (impl_generics, ty_generics, where_clause) = &ast.generics.split_for_impl();
+
     let gen = quote! {
-        impl<'de> ::serde::Deserialize<'de> for #name {
+        impl<'de, #impl_generics> ::serde::Deserialize<'de> for #name #ty_generics
```

Comment:
I think instead of doing this, you need to add `'de` into `impl_generics` above and just use `#impl_generics` here. Not sure how to do that exactly.